rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -1334,6 +1334,64 @@ class DebruijnInterpreter[M[_], F[_]](
       } yield result
   }
 
+  private[this] val toSet: Method = new Method() {
+    def toSet(baseExpr: Expr): M[Par] =
+      baseExpr.exprInstance match {
+        case e: ESetBody =>
+          (e: Par).pure[M]
+        case EListBody(EList(basePs, locallyFree, connectiveUsed, remainder)) =>
+          (ESetBody(
+            ParSet(
+              basePs,
+              connectiveUsed,
+              locallyFree.get.pure[Coeval],
+              remainder
+            )
+          ): Par).pure[M]
+        case other =>
+          MethodNotDefined("toSet", other.typ).raiseError[M, Par]
+      }
+
+    override def apply(p: Par, args: Seq[Par])(implicit env: Env[Par]): M[Par] =
+      for {
+        _ <- if (args.nonEmpty)
+              MethodArgumentNumberMismatch("toSet", 0, args.length).raiseError[M, Unit]
+            else ().pure[M]
+        baseExpr <- evalSingleExpr(p)
+        result   <- toSet(baseExpr)
+      } yield result
+  }
+
+  private[this] val toMap: Method = new Method() {
```

Comment:
This PR would be significantly easier to review if this commit was split into two commits, one adding `toMap` and another adding `toSet`.